operations.go

```diff
@@ -0,0 +1,131 @@
+package tezos
+
+import (
+	"encoding/json"
+	"strconv"
+)
+
+// OperationElem must be implemented by all operation elements
+type OperationElem interface {
+	OperationElemKind() string
+}
+
+// GenericOperationElem is a most generic element type
+type GenericOperationElem struct {
+	Kind string `json:"kind"`
+}
+
+// OperationElemKind implements OperationElem
+func (e *GenericOperationElem) OperationElemKind() string {
+	return e.Kind
+}
+
+// OperationElements is a slice of OperationElem with custom JSON unmarshaller
+type OperationElements []OperationElem
+
+// UnmarshalJSON implements json.Unmarshaler
+func (e *OperationElements) UnmarshalJSON(data []byte) error {
+	var raw []json.RawMessage
+
+	if err := json.Unmarshal(data, &raw); err != nil {
+		return err
+	}
+
+	*e = make(OperationElements, len(raw))
+
+opLoop:
+	for i, r := range raw {
+		var tmp GenericOperationElem
+		if err := json.Unmarshal(r, &tmp); err != nil {
+			return err
+		}
+
+		switch tmp.Kind {
+		case "endorsement":
+			(*e)[i] = &EndorsementOperationElem{}
+
+		default:
+			(*e)[i] = &tmp
+			continue opLoop
```

Comment:
Is the label needed here, given we're just in a one-level loop and can just do `continue`?

Reply:
It isn't, I just added it to avoid confusion as with `break` woud be